blackbox/test_s3.py

```diff
@@ -212,13 +214,96 @@ def test(self):
+                       secure = False)
+
+        with MinioServer() as minio:
+            t = threading.Thread(target=minio.run)
+            t.daemon = True
+            t.start()
+            wait_until(lambda: _is_up('127.0.0.1', 9000))
+
+            client.make_bucket("my.bucket")
+
+            # https://docs.min.io/docs/python-client-api-reference.html#set_bucket_policy
+            # Example anonymous read-write bucket policy.
+            policy = {
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Principal": {"AWS": "*"},
+                        "Action": [
+                            "s3:GetBucketLocation",
+                            "s3:ListBucket",
+                            "s3:ListBucketMultipartUploads",
+                        ],
+                        "Resource": "arn:aws:s3:::my.bucket",
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Principal": {"AWS": "*"},
+                        "Action": [
+                            "s3:GetObject",
+                            "s3:PutObject",
+                            "s3:DeleteObject",
+                            "s3:ListMultipartUploadParts",
+                            "s3:AbortMultipartUpload",
+                        ],
+                        "Resource": "arn:aws:s3:::my.bucket/*",
+                    },
+                ],
+            }
+            client.set_bucket_policy("my.bucket", json.dumps(policy))
+
+            with connect(crate_node.http_url) as conn:
+                c = conn.cursor()
+                c.execute('CREATE TABLE t1 (x int)')
+                c.execute('INSERT INTO t1 (x) VALUES (1), (2), (3)')
+                c.execute('REFRESH TABLE t1')
+
+                c.execute("""
+                    COPY t1 TO DIRECTORY 's3://127.0.0.1:9000/my.bucket/' WITH (protocol = 'http')
+                """)
+                c.execute('CREATE TABLE r1 (x int)')
+                c.execute("""
+                    COPY r1 FROM 's3://127.0.0.1:9000/my.bucket/*' WITH (protocol = 'http')
+                """)
+                c.execute('REFRESH TABLE r1')
+
+                c.execute('SELECT x FROM r1 order by x')
+                result = c.fetchall()
+                self.assertEqual(result, [[1],[2],[3]])
+
+                c.execute("""
+                    COPY r1 FROM ['s3://127.0.0.1:9000/my.bucket/t1_0_.json',
+                    's3://127.0.0.1:9000/my.bucket/t1_1_.json',
+                    's3://127.0.0.1:9000/my.bucket/t1_2_.json',
+                    's3://127.0.0.1:9000/my.bucket/t1_3_.json'] WITH (protocol = 'http')
+                """)
+                c.execute('REFRESH TABLE r1')
+
+                c.execute('SELECT x FROM r1 order by x')
+                result = c.fetchall()
+                self.assertEqual(result, [[1],[1],[2],[2],[3],[3]])
```

Comment:
Maybe limit this to a single copy to/copy from. Based on the test name this is testing the authentication, not the various variants with wildcard import and so on - they're already covered by the other test.

Reply:
Thank you, I would keep a pair of copy from/to since I would like to confirm copy from/to the specified bucket. what do you think?